pyiron_contrib/atomistics/atomistics/job/trainingcontainer.py

```diff
@@ -652,9 +678,33 @@ def to_list(self, filter_function=None):
         num_atoms_list = data_table.number_of_atoms.to_list()
         return structure_list, energy_list, force_list, num_atoms_list
 
+    def iter(self, *arrays, wrap_atoms=True):
+        """
+        Iterate over all structures in this object and all arrays that are defined
+
+        Args:
+            wrap_atoms (bool): True if the atoms are to be wrapped back into the unit cell; passed to
+                               :meth:`.get_structure()`
+            *arrays (str): name of arrays that should be iterated over
+
+        Yields:
+            :class:`pyiron_atomistics.atomistitcs.structure.atoms.Atoms`, arrays: every structure attached to the object and queried arrays
+        """
+        for i in range(self.number_of_structures):
+            s = self._get_structure(frame=i, wrap_atoms=wrap_atoms)
+            vals = [s]
+            for a in arrays:
+                vals.append(
+                    self.get_array(
+                        name=a,
+                        frame=i,
+                    )
+                )
+            yield vals
```

Comment:
```suggestion
yield from zip(self._container.iter_structures(), *(self._container.get_array_ragged(a) for a in arrays))
```
maybe it's a bit too convoluted though.